src/KingSystem/Utils/Byaml/ByamlStringTableIter.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <basis/seadTypes.h>
+
+namespace al {
+
+class ByamlStringTableIter {
+public:
+    explicit ByamlStringTableIter(const u8* data);
+    explicit ByamlStringTableIter(const ByamlStringTableIter& other);
+
+    const char* getString(s32 index) const;
+    s32 findStringIndex(const char* string) const;
+
+    bool isValid() const { return mData != nullptr; }
+
+    const u32* getStringOffsetArray() const {
+        return reinterpret_cast<const u32*>(&mData[mTableOffset]);
+    }
+
+private:
+    const u8* mData;
+
+    static constexpr u32 mTableOffset{4u};
```

Comment:
```suggestion
    static constexpr u32 TableOffset = 4u;
```
(otherwise the name is misleading because it's not actually a non-static member variable, and it's a constant)